testing/matrix-sdk-test/src/test_json/keys_query_sets.rs

```diff
@@ -1,18 +1,230 @@
+use std::default::Default;
+
+use insta::{assert_json_snapshot, with_settings};
 use ruma::{
     api::client::keys::get_keys::v3::Response as KeyQueryResponse, device_id,
-    encryption::DeviceKeys, serde::Raw, user_id, DeviceId, OwnedDeviceId, UserId,
+    encryption::DeviceKeys, serde::Raw, user_id, CanonicalJsonValue, DeviceId, OwnedDeviceId,
+    OwnedUserId, UserId,
 };
 use serde_json::{json, Value};
+use vodozemac::{Curve25519PublicKey, Ed25519SecretKey, Ed25519Signature};
 
 use super::keys_query::{keys_query, master_keys, KeysQueryUser};
 use crate::{
     ruma_response_from_json,
     test_json::keys_query::{device_keys_payload, self_signing_keys},
 };
 
+/// A test helper for building sets of test user data.
+///
+/// Currently, its main purpose is to build `/keys/query` response objects
+/// ([`KeyQueryResponse`]), but that may evolve in future.
+pub struct TestUserDataBuilder {
+    /// The User ID of the user that this test data is about.
+    pub user_id: OwnedUserId,
+
+    /// The user's private master cross-signing key, once it has been set via
+    /// [`TestUserDataBuilder::with_cross_signing_keys`].
+    pub master_cross_signing_key: Option<Ed25519SecretKey>,
+
+    /// The user's private self-signing key, once it has been set via
+    /// [`TestUserDataBuilder::with_cross_signing_keys`].
+    pub self_signing_key: Option<Ed25519SecretKey>,
+
+    /// The user's private user-signing key, once it has been set via
+    /// [`TestUserDataBuilder::with_cross_signing_keys`].
+    pub user_signing_key: Option<Ed25519SecretKey>,
+
+    /// The JSON representation of the user's public master cross-signing key,
+    /// ready for return in the `/keys/query` response.
+    ///
+    /// This starts off as `null`, but is populated with the correct JSON object
```

Comment:
This way we can link to more docs if somebody is interested in what `null` actually means.

```suggestion
    /// This starts off as [`Value::Null`], but is populated with the correct JSON object
```

Reply:
no longer relevant